Core/Frameworks/Flake/Framework.php

```diff
@@ -273,8 +273,12 @@ protected static function initDb() {
         if (defined("BAIKAL_CONTEXT_INSTALL") && (!isset($config['system']['configured_version']) || $config['system']['configured_version'] === BAIKAL_VERSION)) {
             return true;
         }
-        if ($config['database']['mysql'] === true) {
+        # Config key 'mysql' kept for backwards compatibility
+        $legacyMysql = key_exists('mysql', $config['database']) && $config['database']['mysql'] === true;
+        if ($legacyMysql || $config['database']['backend'] === 'mysql') {
```

Comment:
With the legacy configuration, this line now generates a warning when trying to access the new configuration. Because many of our users have server setups that stop the script as soon as there is a warning, this would render Baikal unusable for them

Reply:
Hey :) Thanks for taking the time to give feedback! I'll fix it right away

Reply:
Update: Good catch! I missed that. I could reproduce it, should be fixed now via 577c57d.

Reply:
I have an additional error message on top: `ErrorException: Undefined array key "backend" in /mnt/Entwicklung/htdocs/baikal/Core/Frameworks/BaikalAdmin/Controller/Settings/Database.php:85`. Could you please create a fresh install of Baikal before this PR (both with mysql and sqlite) and then upgrade to this PR to see if everything works? We should make sure that upgrading Baikal does not break existing installs.

Reply:
Sure, doing it now, had a spontaneous appointment

Reply:
Alright, I've fixed the specified line after being able to reproduce the error in `Database.php:85`. For testing I created a proper mysql database inside docker this time and tested all system settings pages / database settings and users / resource pages with both mysql and sqlite configured in the legacy settings format (`mysql = true|false`).

I didn't encounter any other errors and the fix I've applied in `Database.php:85` also translates the legacy configuration to the new "backend" dropdown (either mysql or sqlite for legacy configs) :)

Migrations should be safe with 564ca56.

Reply:
Btw, this now actually allows to "dirty"-migrate by entering the settings web page after bumping the version number manually in `baikal.yaml` and clicking on Save. The action will implicitly upgraded the settings format from `mysql = true|false` to `backend = sqlite|mysql|pgsql`